### PR TITLE
Docs: Flatten the structure of the Framework page in docs.

### PR DESCRIPTION
**Pull request by gorzelinski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Docs: Flatten the structure of the Framework page in docs. Closes https://github.com/cksource/ckeditor5-internal/issues/2671.

---

### Additional information

This PR is pretty big. It required link changes across documentation. Those PRs should be merged just after this one:
https://github.com/cksource/ckeditor5-internal/pull/2707
https://github.com/cksource/collaboration-features/pull/4995

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ For more advanced users, or those who need to integrate CKEditor 5 with their ow
 
 CKEditor 5 builds allow you to quickly and easily initialize one of the many types of editors in your application. At the same time, CKEditor 5 is also a framework for creating custom-made rich text editing solutions.
 
-To find out how to start building your own editor from scratch go to the [CKEditor 5 Framework overview](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/overview.html) section of CKEditor 5 documentation.
```
Comment (dufipl) [inline README.md:92]: This address is actually redirected to https://ckeditor.com/docs/ckeditor5/latest/framework/index.html, which means the new version no longer works. We need to add a new redirect or change this to the actual link.

File: packages/ckeditor5-core/README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ This package implements CKEditor 5's core editor architecture &mdash; a set of c
 
 ## Documentation
 
-For general introduction see the [Overview of CKEditor 5 framework](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/overview.html) guide and then the [core editor architecture guide](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/core-editor-architecture.html).
```
Comment (dufipl) [inline packages/ckeditor5-core/README.md:13]: This address (https://ckeditor.com/docs/ckeditor5/latest/framework/guides/overview.html) is actually redirected to https://ckeditor.com/docs/ckeditor5/latest/framework/index.html, which means the new version no longer works. We need to add a new redirect or change this to the actual link.

File: packages/ckeditor5-engine/README.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ The CKEditor 5 editing engine implements a flexible MVC-based architecture for c
 
 ## Documentation
 
-For a general introduction see the [Overview of CKEditor 5 Framework](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/overview.html) guide and then the [Editing engine architecture guide](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/editing-engine.html).
```
Comment (dufipl) [inline packages/ckeditor5-engine/README.md:25]: This address (https://ckeditor.com/docs/ckeditor5/latest/framework/guides/overview.html) is actually redirected to https://ckeditor.com/docs/ckeditor5/latest/framework/index.html, which means the new version no longer works. We need to add a new redirect or change this to the actual link.

File: packages/ckeditor5-widget/docs/framework/deep-dive/widget-internals.md [hunk 4]
```diff
@@ -67,7 +67,7 @@ widgetTypeAroundPlugin.clearForceDisabled( 'MyApplication' );
 
 Refer to the {@link module:core/plugin~Plugin#clearForceDisabled API documentation} to learn more.
 
-## Exclude DOM events from default handlers
+##  Exclude DOM events from default handlers
```
Comment (dufipl) [inline packages/ckeditor5-widget/docs/framework/deep-dive/widget-internals.md:70]: This extra space looks like not necessary :P

**Review by dufipl — CHANGES_REQUESTED** [review]

File: docs/features/ui-language.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ See the demo of the editor in Spanish:
 </info-box>
 
 <info-box>
-	If you are interested in creating features that can be localized, check out the {@link framework/guides/deep-dive/localization localization guide}.
+	If you are interested in creating features that can be localized, check out the {@link framework/deep-dive/localization localization guide}.
```
Comment (dufipl) [inline docs/features/ui-language.md:27]: This link doesn't work, probably missing a redirect. It should probably point here: `framework/deep-dive/ui/localization.html`.

File: docs/framework/architecture/ui-library.md [hunk 6]
```diff
@@ -9,10 +9,10 @@ The standard UI library of CKEditor 5 is [`@ckeditor/ckeditor5-ui`](https://www.
...
 
-Views use [templates](#templates) to build the UI. They also provide observable interfaces that other features (e.g. {@link framework/guides/architecture/core-editor-architecture#plugins plugins}, {@link framework/guides/architecture/core-editor-architecture#commands commands}, etc.) can use to change the DOM without any actual interaction with the native API.
+Views use [templates](#templates) to build the UI. They also provide observable interfaces that other features (e.g. {@link framework/architecture/core-editor-architecture#plugins plugins}, {@link framework/architecture/core-editor-architecture#commands commands}, etc.) can use to change the DOM without any actual interaction with the native API.
 
 <info-box>
-	All views can be localized using the `locale` instance with which they were created. Check the {@link framework/guides/deep-dive/localization localization guide} to see how to use the `t()` function available in the `locale` instance.
+	All views can be localized using the `locale` instance with which they were created. Check the {@link framework/deep-dive/localization localization guide} to see how to use the `t()` function available in the `locale` instance.
```
Comment (dufipl) [inline docs/framework/architecture/ui-library.md:15]: This link doesn't work, probably missing a redirect. It should probably point here: `framework/deep-dive/ui/localization.html`.

File: docs/framework/package-generator/javascript-package.md [hunk 7]
```diff
@@ -214,11 +214,11 @@ To change the ESLint configuration, edit the [.eslintrc.js](https://github.com/c
...
+To make CKEditor 5 plugins compatible with each other, we needed to introduce certain limitations when importing files from packages. To learn more, visit the {@link installation/advanced/dll-builds DLL guide} and {@link framework/contributing/code-style#dll-builds-ckeditor5-rulesckeditor-imports see a detailed explanation} about the limitations.
 
 ## Translations
 
-Packages created by this tool, just like the entirety of the CKEditor 5 ecosystem, include full support for localization. If you wish to include translations for your package, visit the {@link framework/guides/deep-dive/localization dedicated translation guide} and learn more.
+Packages created by this tool, just like the entirety of the CKEditor 5 ecosystem, include full support for localization. If you wish to include translations for your package, visit the {@link framework/deep-dive/localization dedicated translation guide} and learn more.
```
Comment (dufipl) [inline docs/framework/package-generator/javascript-package.md:221]: This link doesn't work, probably missing a redirect. It should probably point here: `framework/deep-dive/ui/localization.html`.

File: docs/framework/package-generator/typescript-package.md [hunk 8]
```diff
@@ -234,11 +234,11 @@ To change the ESLint configuration, edit the [.eslintrc.js](https://github.com/c
...
+To make CKEditor 5 plugins compatible with each other, we needed to introduce certain limitations when importing files from packages. To learn more, visit the {@link installation/advanced/dll-builds DLL guide} and {@link framework/contributing/code-style#dll-builds-ckeditor5-rulesckeditor-imports see a detailed explanation} about the limitations.
 
 ## Translations
 
-Packages created by this tool, just like the entirety of the CKEditor 5 ecosystem, include full support for localization. If you wish to include translations for your package, visit the {@link framework/guides/deep-dive/localization dedicated translation guide} and learn more.
+Packages created by this tool, just like the entirety of the CKEditor 5 ecosystem, include full support for localization. If you wish to include translations for your package, visit the {@link framework/deep-dive/localization dedicated translation guide} and learn more.
```
Comment (dufipl) [inline docs/framework/package-generator/typescript-package.md:241]: This link doesn't work, probably missing a redirect. It should probably point here: `framework/deep-dive/ui/localization.html`.

File: docs/installation/advanced/csp.md [hunk 9]
```diff
@@ -41,7 +41,7 @@ Some CSP directives have an impact on certain rich-text editor features. Here is
 
 	**Note**: Use the more strict `img-src 'self'` if all images in the editor content are hosted from the same domain and you do **not** want to enable the {@link features/media-embed media embed} and {@link features/paste-from-word paste from Word} features.
 * `style-src 'self' 'unsafe-inline'`: `'unsafe-inline'` is necessary for:
-	* webpack's [style-loader](https://github.com/webpack-contrib/style-loader) to load the {@link framework/guides/theme-customization#styles-processing-and-bundling editor UI styles}.
+	* webpack's [style-loader](https://github.com/webpack-contrib/style-loader) to load the {@link framework/theme-customization#styles-processing-and-bundling editor UI styles}.
```
Comment (dufipl) [inline docs/installation/advanced/csp.md:44]: This link doesn't work, probably missing a redirect. It should probably point here: `framework/deep-dive/ui/theme-customization.html`.

File: README.md (hunk 1 above)
Comment (godai78) [reply README.md:92]: But, isn't `framework/index.html` a correct link now (and used to be anyway)?

**Review by godai78 — APPROVED** [review]
LGTM

**Comment by godai78** [comment]
Please double-check the final `stable` merge ([made to resolve conflicts](https://github.com/ckeditor/ckeditor5/pull/13465/commits/0a8cd5d7c326d9127e033aa9d8a773b076679212)) before committing this to the codebase.

File: docs/umberto.json [hunk 10]
```diff
@@ -121,13 +121,47 @@
 		"features/spell-checker.html": "features/spelling-and-grammar-checking.html",
 		"features/collaboration/comments/comments-api.html": "api/comments.html",
 		"features/ckbox.html": "features/images/image-upload/ckbox.html",
-		"framework/guides/ui/document-editor.html": "framework/guides/deep-dive/ui/document-editor.html",
```
Comment (godai78) [inline docs/umberto.json:124]: Are these replaced with new links? Because we should not remove them totally.

Comment (gorzelinski) [reply docs/umberto.json:126]: Yes, I added new redirects and updated the old ones to avoid double redirects.

Comment (godai78) [reply docs/umberto.json:126]: FIne, thanks!

File: README.md (hunk 1 above)
Comment (dufipl) [reply README.md:92]: Right, now it works, so let's discard this comment.

**Review by dufipl — APPROVED** [review]
LGTM 👍

**Pull request merged**